src/app/containers/PageHandlers/withLoading/index.jsx

```diff
@@ -10,14 +12,20 @@ const LoadingMain = styled.main`
 
 const WithLoading = Component => {
   const LoadingContainer = ({ loading, ...props }) => {
-    const [showLoading, setShowLoading] = useState(false);
+    const loadingMessageRef = useRef();
+
     useEffect(() => {
       if (loading) {
         timeout = setTimeout(() => {
-          setShowLoading(true);
+          if (loadingMessageRef.current) {
+            loadingMessageRef.current.focus();
+          }
         }, 500);
       }
-      return () => clearTimeout(timeout);
+
+      return () => {
```

Comment:
Dumb Q - is this like a `componentWillUnmount` - to stop the timeout function running when the component is no longer mounted?

Reply:
Yeah. Not dumb btdubs

Reply:
yeh exactly. if the component unmounts in less than 500ms then it would try to focus on a ref that doesn't exist and throw an error. it's called a cleanup function